app/models/comment.php

```diff
@@ -54,11 +158,51 @@ public function create(Thread $thread)
         }
 
         $db = DB::conn();
-        $db->insert('comment', array(
-            'thread_id' => $thread->id,
-            'user_id'   => $this->user_id,
-            'body'      => $this->body
-        ));
+        $db->insert(
+            'comment',
+            array(
+                'thread_id' => $thread->id,
+                'user_id'   => $this->user_id,
+                'body'      => $this->body,
+                'created_at'=> null
+            )
+        );
     }
 
+    public function update()
+    {
+        if (!$this->validate()) {
+            throw new ValidationException('Invalid comment.');
+        }
+
+        $db = DB::conn();
+        $db->update(
+            self::TABLE_NAME,
+            array('body' => $this->body),
+            array('id' => $this->id)
+        );
+    }
+
+    public function delete()
+    {
+        $db = DB::conn();
+        $db->query(sprintf('DELETE FROM %s WHERE id=?', self::TABLE_NAME), array($this->id));
+    }
+
+    public function isOwnedBy($user)
```

Comment:
Since the return is boolean po, please rename to ie. isOwner() or isAuthor() :bow: